app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -606,16 +606,19 @@ public void handleIntent(@NonNull final Intent intent) {
         if (intent.hasExtra(PLAYBACK_QUALITY)) {
             setPlaybackQuality(intent.getStringExtra(PLAYBACK_QUALITY));
         }
-
-        // Resolve append intents
-        if (intent.getBooleanExtra(APPEND_ONLY, false) && playQueue != null) {
+        // Resolve enqueue intents
+        if (intent.getBooleanExtra(ENQUEUE, false)
+                && playQueue != null) { /* Enqueue = true && playQueue not empty*/
             final int sizeBeforeAppend = playQueue.size();
             playQueue.append(newQueue.getStreams());
 
-            if ((intent.getBooleanExtra(SELECT_ON_APPEND, false)
-                    || currentState == STATE_COMPLETED) && newQueue.getStreams().size() > 0) {
-                playQueue.setIndex(sizeBeforeAppend);
-            }
+            return;
+        // Resolve enqueue next intents
+        } else if (intent.getBooleanExtra(ENQUEUE_NEXT, false)
+                && playQueue != null) { /* Enqueue next = true && playQueue not empty*/
```

Comment:
`/* Enqueue next = true && playQueue not empty*/`
Mmmh, this comment seems redundant to me. Also, this is not checking whether the playqueue is not empty, but just whether is it is non null, i.e. whether a queue was provided at all.